generator_process/actions/detect_seamless/__init__.py

```diff
@@ -0,0 +1,165 @@
+        return False
+
+    @classmethod
+    def _missing_(cls, value):
+        if isinstance(value, str):
+            if value == '':
+                return cls.OFF
+            for e in cls:
+                if e.id == value or e.text == value:
+                    return e
+            raise ValueError(f'no {cls.__name__} with id {repr(id)}')
+        elif isinstance(value, (tuple, list)) and len(value) == 2:
+            for e in cls:
+                if e.x == value[0] and e.y == value[1]:
+                    return e
+            raise ValueError(f'no {cls.__name__} with x {value[0]} and y {value[1]}')
+        elif value is True:
+            return cls.BOTH
+        elif value is False:
+            return cls.OFF
+        elif value is None:
+            return cls.AUTO
+        raise TypeError(f'expected str, bool, tuple[bool, bool], or None, got {repr(value)}')
+
+    def bpy_enum(self, *args):
+        return self.id, self.text, *args
+
+
+def detect_seamless(self, image: NDArray) -> SeamlessAxes:
+    import os
+    import torch
+    from torch import nn
+
+    if image.shape[0] < 8 or image.shape[1] < 8:
+        return SeamlessAxes.OFF
+
+    model = getattr(self, 'detect_seamless_model', None)
+    if model is None:
+        state_npz = np.load(os.path.join(os.path.dirname(__file__), 'model.npz'))
+        state = {k: torch.tensor(v) for k, v in state_npz.items()}
+
+        class SeamlessModel(nn.Module):
+            def __init__(self):
+                super(SeamlessModel, self).__init__()
+                self.conv = nn.Sequential(
+                    nn.Conv2d(3, 64, kernel_size=3, stride=1, padding=1),
+                    nn.Dropout(.2),
+                    nn.PReLU(64),
+                    nn.Conv2d(64, 16, kernel_size=3, stride=1, padding=1),
+                    nn.Dropout(.2),
+                    nn.PReLU(16),
+                    nn.Conv2d(16, 64, kernel_size=8, stride=4, padding=0),
+                    nn.Dropout(.2),
+                    nn.PReLU(64),
+                    nn.Conv2d(64, 64, kernel_size=(1, 3), stride=1, padding=0),
+                    nn.Dropout(.2)
+                )
+                self.gru = nn.GRU(64, 32, batch_first=True)
+                self.fc = nn.Linear(32, 1)
+
+            def forward(self, x: torch.Tensor):
+                if len(x.size()) == 3:
+                    x = x.unsqueeze(0)
+                x = self.conv(x)
+                h = torch.zeros(self.gru.num_layers, x.size()[0], self.gru.hidden_size,
+                                dtype=x.dtype, device=x.device)
+                x, h = self.gru(x.squeeze(3).transpose(2, 1), h)
+                return torch.tanh(self.fc(x[:, -1]))
+
+        model = SeamlessModel()
+        model.load_state_dict(state)
+        model.eval()
+        setattr(self, 'detect_seamless_model', model)
+
+    if torch.cuda.is_available():
+        device = 'cuda'
+    elif torch.backends.mps.is_available():
+        device = 'mps'
```

Comment:
The model does not output correct results on `mps`. I poked around a bit and it seems it seems to deviate from the CPU results most through the `GRU` layer, but I think using CPU is fine as it seems to run just as fast.
```suggestion
        device = 'cpu'
```